clinica/pydra/dwi/_commons/_workflows.py

```diff
@@ -0,0 +1,153 @@
+        Name of the pipeline. Default='eddy_fsl'.
+
+    Returns
+    -------
+    Workflow :
+        The Nipype workflow.
+        This workflow has the following inputs:
+            - "dwi_filename": The path to the DWI image
+            - "b_vectors_filename": The path to the associated B-vectors file
+            - "b_values_filename": The path to the associated B-values file
+            - "in_mask": The path to the mask image to be provided to Eddy
+              This will be used only if compute_mask is False.
+            - "image_id": Prefix to be used for output files
+            - "field": The path to the field image to be used by Eddy
+            - "reference_b0": The path to the reference B0 image
+              This will be used to compute the mask only if compute_mask is True.
+            - "total_readout_time": The total readout time extracted from JSON metadata
+            - "phase_encoding_direction": The phase encoding direction extracted from JSON metadata
+
+        And the following outputs:
+            - "out_parameter": Path to the file storing the output parameters
+            - "out_corrected": Path to the corrected image
+            - "out_rotated_bvecs": Path to the file holding the rotated B-vectors
+    """
+    from pydra.tasks import fsl
+
+    from ._tasks import generate_acq_file_task, generate_index_file_task
+
+    wf = Workflow(
+        name, input_spec=_build_eddy_fsl_input_specs(image_id, field, compute_mask)
+    )
+
+    bet_config = {
+        "name": "mask_reference_b0",
+        "fractional_intensity_threshold": 0.3,
+        "save_brain_mask": True,
+        "with_robust_brain_center_estimation": True,
+    }
+    if compute_mask:
+        bet_config["input_image"] = wf.lzin.reference_b0
+
+    wf.add(fsl.BET(**bet_config))
+
+    wf.add(
+        generate_acq_file_task(
+            dwi_filename=wf.lzin.dwi_filename,
+            fsl_phase_encoding_direction=wf.lzin.phase_encoding_direction,
+            total_readout_time=wf.lzin.total_readout_time,
+            image_id=wf.lzin.image_id if image_id else None,
+        )
+    )
+
+    wf.add(
+        generate_index_file_task(
+            b_values_filename=wf.lzin.b_values_filename,
+            image_id=wf.lzin.image_id if image_id else None,
+        )
+    )
+
+    eddy_config = {
+        "name": "eddy_fsl",
+        "replace_outliers": True,
+        "bvec_file": wf.lzin.b_vectors_filename,
+        "bval_file": wf.lzin.b_values_filename,
+        "input_image": wf.lzin.dwi_filename,
+        "encoding_file": wf.generate_acq_file_task.lzout.out_file,
+        "index_file": wf.generate_index_file_task.lzout.out_file,
+        "brain_mask": wf.mask_reference_b0.lzout.brain_mask
+        if compute_mask
+        else wf.lzin.in_mask,
+    }
+    if image_id:
+        eddy_config["output_basename"] = wf.lzin.image_id
+    if field:
+        eddy_config["fieldmap_image"] = wf.lzin.field
+
+    wf.add(fsl.Eddy(**eddy_config))
```

Comment:
Another reason why I would discourage the convention of passing a dictionary is that there are talks in adjusting the syntax to separate task specific arguments and runtime arguments (name, cache_dir, ...).

I won't reject it if you prefer it, but thought I should mention it now that it's being discussed.